src/pspm_pupil_pp.m

```diff
@@ -147,19 +151,36 @@
   [sts2, eye2] = pspm_find_eye(data_combine.header.chantype);
   if (sts1 < 1 || sts2 < 1), return, end
   if sum(strcmp([eye1, eye2], {'lr', 'rl'})) < 1
-    warning('ID:invalid_input', 'options.channel and options.channel_combine must specify left and right eyes');
+    warning('ID:invalid_input', ...
+      'options.channel and options.channel_combine must specify left and right eyes.');
     return;
   end
   if data.header.sr ~= data_combine.header.sr
-    warning('ID:invalid_input', 'options.channel and options.channel_combine data have different sampling rate');
+    warning('ID:invalid_input', ...
+      'options.channel and options.channel_combine data have different sampling rate.');
     return;
   end
   if ~strcmp(data.header.units, data_combine.header.units)
-    warning('ID:invalid_input', 'options.channel and options.channel_combine data have different units');
+    warning('ID:invalid_input', ...
+      'options.channel and options.channel_combine data have different units.');
     return;
   end
   if numel(data.data) ~= numel(data_combine.data)
-    warning('ID:invalid_input', 'options.channel and options.channel_combine data have different lengths');
+    warning('ID:invalid_input', ...
+      'options.channel and options.channel_combine data have different lengths.');
+    return;
+  end
+  % Check if one channel is dominated by NaNs
```

Comment:
I think what the function should do is slightly different:
- if only one channel is provided, then just give a warning but continue processing
- if two channels are provided, then give a warning and continue processing with the better channel only, do not combine 
The problem the user had initially reported was that if one channel is bad, then the other one will be compromised by the combination.

Reply:
I understand, thank you @dominikbach for the input.
Now I have updated the logic.
1. If the original `data` channel is bad (has too many NaNs that are more than the cut-off), give a warning but continue processing. This is not related to the `data_combine` channel. The behaviour will `not` terminate the function.
2. If the `data_combine` channel is bad (has too many NaNs that are more than the cut-off), give a warning *and* do not combine, but continue processing with the original data channel, this is equivalent to "data_combine is not specified". The behaviour will `not` terminate the function.
3. If the `data_combine` channel is not specified, in this case it will be defined as empty (`[]`), the default value. This will continue processing the original data channel. The behaviour will `not` terminate the function.

My understanding is this can provide the expected behaviours. Please forgive me if I am not correct. Thank you.